OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -17,25 +19,40 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class Land : Activity
 	{
-		readonly Target target;
 		readonly Aircraft aircraft;
 		readonly WVec offset;
+		readonly int desiredFacing;
+		readonly CPos[] clearCells;
 
+		Target target;
+		WPos targetPosition;
+		WPos w1, w2, w3;
+		CPos landingCell;
 		bool landingInitiated;
-		bool soundPlayed;
+		bool finishedApproach;
 
-		public Land(Actor self, Target t, WVec offset)
+		public Land(Actor self, Target target = new Target(), int facing = -1, WVec offset = new WVec(), CPos[] clearCells = null)
```

Comment:
I know that my complaining over this must be getting really annoying now, but my motivation comes from dealing with an endless series of bugs like #15520 and #16473 caused by bad/missing handling of default values (`CPos.Zero`), and from all the regressions caused by #15866 and the massive scope creep required to fix them because none of us appreciated all the subtleties of target handling before/during review.

Something like 5a531cf913 or 61dfaa1aba may not be particularly elegant, but sticking to the same boring code patterns used everywhere else helps reduce future "WTF?!?"s and unexpected regressions the next time this code needs to be refactored.

Of course the real issue here is that `Target` is no longer really fit for purpose, and needs a general overhaul to properly represent our newer usecases:
* #16270
* Representing actors that we last saw at position X but are not currently visible
* Handling actors that change state - frozen/visible, transforms, ownership, etc
* Representing "no target" as distinct from "target which is no longer valid" - and making "no target" the default value

This is a major project in its own right, and IMO we should stay well clear of it until after the activity rework is finished and stable.

Reply:
I'm a bit confused. Introducing `Target.IsNone` is supposed to alleviate these very issues. Specifically:

> Representing "no target" as distinct from "target which is no longer valid" - and making "no target" the default value

is exactly what I have done here. So I'm not sure what you're asking. Is there anything in particular that is wrong with my implementation? Or do you mean to say that the general idea is right, but should be handled in a separate PR? Or do I misunderstand completely?

Reply:
Sorry for the confusion. What I meant was that the general idea is good, but that we really need to implement this as a first-class `TargetType` and use it everywhere that it is relevant, instead of bolting it on as a workaround for just these specific activities.

Reply:
Okay, that makes sense. My idea was that it could be introduced more gradually. Because it doesn't change anything for use-cases that do not actually call the introduced method, it is guaranteed to not cause regressions in code that I didn't touch. It could then be used in other places where it makes sense at a later stage. However, you may be right that reworking `Target` in a more holistic fashion is better.

Reply:
Reverted the addition of `Target.IsNone` and reworked based on the pattern of 5a531cf